cmd/dlt/admin/cmd.go

```diff
@@ -45,85 +99,58 @@ func run(cmd *cobra.Command, _ []string) {
 		// delete `cluster-admin` user from the HTPasswd IDP
-		r.Reporter.Debugf("Deleting user '%s' identity provider on cluster '%s'", idp.ClusterAdminUsername, clusterKey)
-		err = r.OCMClient.DeleteUser(cluster.ID(), "cluster-admins", idp.ClusterAdminUsername)
+		r.Reporter.Debugf("Deleting user '%s' from cluster-admins group on cluster '%s'",
+			idp.ClusterAdminUsername, r.ClusterKey)
+		err = r.OCMClient.DeleteUser(clusterID, "cluster-admins", idp.ClusterAdminUsername)
 		if err != nil {
 			r.Reporter.Errorf("Failed to delete '%s' user from cluster-admins groups of cluster '%s': %s",
-				idp.ClusterAdminUsername, clusterKey, err)
-			os.Exit(1)
-		}
-
-		//delete now the cluster-admin user from the htpasswd idp
-		users, err := r.OCMClient.GetHTPasswdUserList(cluster.ID(), htpasswdIDP.ID())
-		if err != nil {
-			r.Reporter.Errorf("Failed to list htpasswd idp users of cluster '%s': %s",
-				clusterKey, err)
+				idp.ClusterAdminUsername, r.ClusterKey, err)
 			os.Exit(1)
 		}
-		for _, user := range users.Slice() {
-			if user.Username() == idp.ClusterAdminUsername {
-				err := r.OCMClient.DeleteHTPasswdUser(idp.ClusterAdminUsername, cluster.ID(), htpasswdIDP)
-				if err != nil {
-					r.Reporter.Errorf("Failed to delete '%s' user from htpasswd idp users list of cluster '%s': %s",
-						idp.ClusterAdminUsername, clusterKey, err)
-					os.Exit(1)
-				}
-
-			}
-		}
 
-		users, err = r.OCMClient.GetHTPasswdUserList(cluster.ID(), htpasswdIDP.ID())
-		if err != nil {
-			r.Reporter.Errorf("Failed to list htpasswd idp users of cluster '%s': %s",
-				clusterKey, err)
-			os.Exit(1)
+		var deletionStrategy DeleteAdminUserStrategy = &DeleteUserAdminFromIDP{}
+		if wasAdminCreatedUsingOldROSA(r, identityProvider) {
+			deletionStrategy = &DeleteAdminIDP{}
 		}
+		deletionStrategy.deleteAdmin(r, identityProvider)
 
-		htpasswdIdentityProvider, ok := htpasswdIDP.GetHtpasswd()
-		if !ok {
-			r.Reporter.Errorf("Failed to get htpasswd idp of cluster '%s': %s",
-				clusterKey, err)
-			os.Exit(1)
-		}
+		r.Reporter.Infof("Admin user '%s' has been deleted from cluster '%s'", idp.ClusterAdminUsername, r.ClusterKey)
+	}
+}
 
-		if users.Len() == 0 && htpasswdIdentityProvider.Username() == "" {
-			//delete the idp as users list is empty
-			err := r.OCMClient.DeleteIdentityProvider(cluster.ID(), htpasswdIDP.ID())
-			if err != nil {
-				r.Reporter.Errorf("Failed to delete htpasswd idp '%s' of cluster '%s': %s",
-					htpasswdIDP.ID(), clusterKey, err)
-				os.Exit(1)
-			}
-		}
-		r.Reporter.Infof("Admin user '%s' has been deleted from cluster '%s'", idp.ClusterAdminUsername, clusterKey)
+func wasAdminCreatedUsingOldROSA(r *rosa.Runtime, identityProvider *cmv1.IdentityProvider) bool {
+	htpasswdIdp, ok := identityProvider.GetHtpasswd()
+	if !ok {
+		r.Reporter.Errorf("Failed to get htpasswd idp for cluster '%s'", r.Cluster.ID())
+		os.Exit(1)
 	}
+	return htpasswdIdp.Username() == "cluster-admin"
```

Comment:
do we define the literal `cluster-admin` somewhere? we should probably use that.